### ISSUE TITLE
Sanic `Request` object is falsey

### ISSUE THREAD

**Issue by logannc** [channelcat/sanic#1221]
```python
@app.route('/myroute')
async def someroute(request):
    if request:
        return 'some data'
    raise Exception("Woops")
```

This code will raise the exception because `bool(request)` is `False`.

**Comment by Tronic** [comment]
Can anyone comment again *why* request should be truthy when connected to transport? I'm wondering what the heck that `__bool__` is for.

**Comment by logannc** [comment]
Not sure if this is still the case, but at the time this issue was opened, `ContentRangeHandler` took a request object as a parameter. So, in handler I was writing that optionally supported the `Range` header, I needed to pass in a request object. This object was an optional parameter as it is not needed if you do not intend to process the `Range` header. Thus, it looked a bit like:

```python
if _request: # if _request is not None
	try:
		range = ContentRangeHandler(_request, stat)
	except (...):
	    pass
```

This failed because `_request` was _always_ falsey (due to it being a subclass of `dict`, I believe, I don't actually recall). I worked around it by testing for `None`, but being truthy is slightly more pythonic.
